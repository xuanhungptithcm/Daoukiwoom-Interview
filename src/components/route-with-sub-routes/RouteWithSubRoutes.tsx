/**
 *
 * RouteWithSubRoutes
 *
 */

import React from "react";
import { Route } from "react-router-dom";
import { IRouteProps } from "../../routes/type";
import PrivateRoute from "../routing/PrivateRoute";

function RouteWithSubRoutes(route: IRouteProps) {
  return route.requiredAuthen ? (
    <PrivateRoute {...route} />
  ) : (
    <Route
      exact={route.exact}
      path={route.path}
      render={(routeProps) => {
        if (!routeProps.match) return null;
        if (route.component) {
          return (
            // pass the sub-routes down to keep nesting
            <route.component {...routeProps} routes={route?.routes} />
          );
        }
        return null;
      }}
    />
  );
}

export default RouteWithSubRoutes;
