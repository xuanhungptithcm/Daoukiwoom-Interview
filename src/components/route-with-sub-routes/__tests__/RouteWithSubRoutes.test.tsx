/**
 * @jest-environment jsdom
 */

import { shallow } from "enzyme";
import RouteWithSubRoutes from "../RouteWithSubRoutes";

describe("RouteWithSubRoutes", () => {
  test("should properly render", () => {
    const routes: any = [
      {
        component: "A",
        path: "/a",
        key: "1",
        requiredAuthen: false,
        exact: false,
        isAuthorized: false,
      },
      {
        component: "B",
        route: "/b",
        key: "2",
        requiredAuthen: false,
        exact: false,
        isAuthorized: false,
      },
    ];
    expect(shallow(<RouteWithSubRoutes {...routes} />)).toMatchSnapshot();
  });
});
