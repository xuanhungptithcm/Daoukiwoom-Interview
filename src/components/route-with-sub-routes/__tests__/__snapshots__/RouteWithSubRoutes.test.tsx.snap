// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteWithSubRoutes should properly render 1`] = `
<Route
  render={[Function]}
/>
`;
